crates/wallet/src/wallet/tx_builder.rs

```diff
@@ -274,26 +274,28 @@ impl<'a, Cs> TxBuilder<'a, Cs> {
     /// These have priority over the "unspendable" utxos, meaning that if a utxo is present both in
     /// the "utxos" and the "unspendable" list, it will be spent.
     pub fn add_utxos(&mut self, outpoints: &[OutPoint]) -> Result<&mut Self, AddUtxoError> {
-        {
-            let wallet = &mut self.wallet;
-            let utxos = outpoints
-                .iter()
-                .map(|outpoint| {
-                    wallet
-                        .get_utxo(*outpoint)
-                        .ok_or(AddUtxoError::UnknownUtxo(*outpoint))
-                })
-                .collect::<Result<Vec<_>, _>>()?;
-
-            for utxo in utxos {
-                let descriptor = wallet.public_descriptor(utxo.keychain);
-                let satisfaction_weight = descriptor.max_weight_to_satisfy().unwrap();
-                self.params.utxos.push(WeightedUtxo {
-                    satisfaction_weight,
-                    utxo: Utxo::Local(utxo),
-                });
-            }
-        }
+        outpoints
+            .iter()
+            .map(|outpoint| {
+                self.wallet
+                    .get_utxo(*outpoint)
+                    .ok_or(AddUtxoError::UnknownUtxo(*outpoint))
+                    .map(|output| {
+                        (
+                            *outpoint,
+                            WeightedUtxo {
+                                satisfaction_weight: self
+                                    .wallet
+                                    .public_descriptor(output.keychain)
+                                    .max_weight_to_satisfy()
+                                    .unwrap(),
+                                utxo: Utxo::Local(output),
+                            },
+                        )
+                    })
+            })
+            .collect::<Result<HashMap<OutPoint, WeightedUtxo>, AddUtxoError>>()
```

Comment:
Nit: Can we collect this into a variable first? I think it'll make it easier to read.